homeassistant/components/arcam_fmj/device_trigger.py

```diff
@@ -31,9 +31,8 @@
 )
 
 
-async def async_get_triggers(
-    hass: HomeAssistant, device_id: str
-) -> list[dict[str, str]]:
+@callback
+def async_get_triggers(hass: HomeAssistant, device_id: str) -> list[dict[str, str]]:
```

Comment:
A possible downside of this api change is that we stop integrations from being able to await things inside this function. We don't seem to need that yet, but it is a limitation that we introduce.

Reply:
Originally, they all needed to be `async` because of the `device_registry` (`async_get_registry` versus `async_get`)
Once that was updated, the only component that really needed a refactor was `nest`, and that was solved with a minor refactor in #72170

We can also continue to handle both, and never deprecate the use of coroutines.

Reply:
I'm happy with that.
The plan was to handle both until core was updated, but if we want to continue handling both I can remove the comments from #72147

Reply:
I have created a separate PR to remove the deprecation: #72904
I think this PR still makes sense though...

Reply:
It seems the view is to revert #72147, despite the fact that none of the 6 methods do any awaiting in any component ?

Reply:
I've marked the reversal PR as ready for review. I'll close this one when it's merged

Reply:
@epenet is right, the reason that the device automation is based on coroutine functions is that when it was implemented we didn't load the registries at startup; they had to be awaited. This added a lot of complications everywhere in the device automation framework, for example also schema validation had to rely on coroutine functions.

Is it reasonable that fetching a list of automations for a device or entity actually needs to speak to that device or entity instead of deriving the information from data which is stored in the registries or in the state machine?

Reply:
I talked with epenet. We decided to keep the async tag so we don't hinder future integrations that have fancy ways of doing triggers. 

Note that for future discussions about these 2 types of async functions, let's make sure we are consistent in terminology. The two types of async functions are called `callback` and `async`. The term `sync` is for functions that do blocking I/O and need to run in the executor.

Reply:
Asynchronous functions are called coroutine functions. `async` is not a good term as callbacks are run in what we call async context, inside the event loop.